src/modules/staking/services/staking.compute.service.ts

```diff
@@ -211,4 +213,76 @@ export class StakingComputeService {
                   .dividedBy(constantsConfig.MAX_PERCENT)
                   .toFixed();
     }
+
+    async computeOptimalCompoundFrequency(
+        stakeAddress: string,
+        positionAmount: string,
+        timeHorizon: number,
+    ): Promise<OptimalCompoundModel> {
+        const [apr, farmingToken] = await Promise.all([
+            this.stakeFarmAPR(stakeAddress),
+            this.stakingService.getFarmingToken(stakeAddress),
+        ]);
+
+        const denominatedAmount = denominateAmount(
+            positionAmount,
+            farmingToken.decimals,
+        ).toNumber();
+
+        const farmingTokenPrice = await this.tokenGetter.getDerivedEGLD(
+            farmingToken.identifier,
+        );
+        const egldPriceFarmingToken = new BigNumber(1).dividedBy(
+            farmingTokenPrice,
+        );
+        const transactionFeesInFarmingToken = new BigNumber(
+            constantsConfig.COMPOUND_TRANSACTION_FEE,
+        )
+            .multipliedBy(egldPriceFarmingToken)
+            .toNumber();
+
+        // Express compound iterations in minutes
+        const compoundIterations =
+            constantsConfig.COMPOUND_INTERVAL_DAYS * 24 * 60;
+
+        let optimalCompoundIterations = 0;
+        let optimalProfit = 0;
+
+        for (let iterator = 1; iterator < compoundIterations; iterator += 1) {
```

Comment:
If `COMPOUND_INTERVAL_DAYS = 365`, here we could iterate over `compoundIterations = 365 * 24 * 60 = 525600` elements. Wouldn't it make computations expansive or is it fine? Alternatively, we could reduce the cost by considering only 365 possible compounding times (1 per day).

Reply:
done